src/main/java/com/github/joschi/jadconfig/repositories/SystemPropertiesRepository.java

```diff
@@ -46,6 +49,11 @@ public String read(String name) {
         return System.getProperty(prefix + name);
     }
 
+    @Override
+    public Collection<String> readNames(String prefix) {
+        return System.getProperties().stringPropertyNames().stream().filter(key -> key.startsWith(prefix)).collect(Collectors.toSet());
```

Comment:
This will not work as there are two prefixes to be considered.

Reply:
Repositories handle only one prefix at a time to make the api cleaner and simpler to understand and implement. The actual iteration over several prefixes happens here: https://github.com/Graylog2/JadConfig/pull/130/files#diff-c380f73f186dbedc110e787e2edc02822155949e4ac5d68d706dd0c417e9fefdR238

Reply:
fixed, thanks!